core/src/iosMain/kotlin/com/revenuecat/purchases/kmp/PurchasesDelegate.ios.kt

```diff
@@ -45,3 +46,20 @@ private class PurchasesDelegateWrapper(val wrapped: PurchasesDelegate) :
     }
 
 }
+
+/**
+ * On the iOS platform side, the backing field of `Purchases.delegate`, `Purchases.privateDelegate`,
+ * is a `weak var`. On the multiplatform side, we wrap the actual delegate in
+ * `PurchasesDelegateWrapper` to conform to `RCPurchasesDelegateProtocol`. Since that is a private
+ * implementation detail, the only reference held to our wrapper is the `weak var`. This means that
+ * it gets deallocated the first chance it gets. For this reason, we keep a strong reference to our
+ * `PurchasesDelegateWrapper` here. This matches the Android behavior, which keeps a strong
+ * reference on the platform side already.
+ *
+ * **Note**: we cannot make our `PurchasesDelegateWrapper` an `object` directly, as objects cannot
+ * extend `NSObject`. See also
+ * [KT67930](https://youtrack.jetbrains.com/issue/KT-67930/Getting-Crash-while-building-KMM-project-with-XCode-15.3#focus=Comments-27-9796215.0-0)
+ */
+private object PurchasesDelegateStrongReference {
+    var delegate: RCPurchasesDelegateProtocol? = null
```

Comment:
Would it be possible to just hold this inside the `Purchases.ios` class instead? Just to make sure if it gets dereferenced (without calling the `close` function, we also clear it. (honestly this is an edge case, since usually `Purchases` would live forever, but just in case)

Reply:
Ah yes that makes way more sense haha! https://github.com/RevenueCat/purchases-kmp/pull/214/commits/1810b44490c0b0941836fe36c24bf122fbe5703e